package/MDAnalysis/lib/util.py

```diff
@@ -222,32 +223,48 @@ def get_reader_for(filename, format=None):
     filename
         filename of the input trajectory or coordinate file
         Also can handle a few special cases, see notes below.
+    format
+        Define the desired format.  Can be a string to request
+        a given Reader.
+        If a class is passed, it will be assumed that this is
+        a Reader and will be returned.
 
     Returns
     -------
     A Reader object
 
+
+    Raises
+    ------
+    ValueError
+        If no appropriate Reader is found
+
+
     Notes
     -----
     There are a number of special cases that can be handled:
-      If *filename* is a numpy array, MemoryReader is returned.
-      If *filename* is an MMTF object, MMTFReader is returned.
-      If *filename* is an iterable of filenames, ChainReader is returned.
+      If `filename` is a numpy array, MemoryReader is returned.
+      If `filename` is an MMTF object, MMTFReader is returned.
+      If `filename` is an iterable of filenames, ChainReader is returned.
 
     Automatic detection is disabled when an explicit *format* is
     provided.
     """
+    # check if format is actually a Reader
+    if inspect.isclass(format):
+        return format
+
     # Checks for specialised formats
     # memoryreader slurps numpy arrays
     if isinstance(filename, np.ndarray):
-        format = 'memory'
+        format = 'MEMORY'
     # mmtf slurps mmtf object
     elif isinstance(filename, mmtf.MMTFDecoder):
-        format = 'mmtf'
+        format = 'MMTF'
     # chainreader for iterables
     # order is important here, np.array is iterable but gets stolen above
     elif iterable(filename):
-        format = 'chain'
+        format = 'CHAIN'
 
     # else let the guessing begin!
     if format is None:
```

Comment:
It should have been `if format is None:` before all guesses, ie only guess if `format` isn't given.